### DISCUSSION TITLE
Vague message "Deprecated features are being used"

### DISCUSSION THREAD

**Discussion by ioutkine** [rabbitmq/rabbitmq-server#11350] [Questions]
Following the upgrade to version 3.13, the RabbitMQ console has begun to display a message stating, 'Deprecated features are being used.' However, the message does not specify which features are considered deprecated. Clicking on the 'Learn More' link leads to a page that fails to provide further clarification, as it does not even mention the term 'deprecated.' Any assistance in identifying these deprecated features would be greatly appreciated.

![image](https://github.com/rabbitmq/rabbitmq-server/assets/25414527/45894550-7545-4384-8740-ea9350530d17)

**Comment by ioutkine** [comment]
In server logs I see following error
Deprecated features: `management_metrics_collection`: Feature `management_metrics_collection` is deprecated.

But I cannot find any documentation where recommended settings are suggested.

**Comment by ansd** [comment]
Documentation is currently missing. The only documentation currently exists in https://github.com/rabbitmq/rabbitmq-server/pull/7390

**Reply by ioutkine** [reply]
Means it is too early to worry? if management_metrics_collection is going to be deprecated then what will be the alternative?

**Reply by gomoripeti** [reply]
You don't need to worry but it is good to be prepared. Fetching metrics via the Prometheus endpoint is the newer and preferred way (https://www.rabbitmq.com/docs/prometheus). The "deprecated features" functionality is just the implementation of something that was announced a few years ago already https://www.rabbitmq.com/blog/2021/08/21/4.0-deprecation-announcements. That blog post is also a good source to find out a bit more about each feature and alternatives.

**Reply by ioutkine** [reply]
Actually it seems I have broken the basic monitoring in the management console. Even if I add 
deprecated_features.permit.management_metrics_collection = true
to the config, I still see the "deprecated feature" warning but I do not see connection, channel details any longer:
![image](https://github.com/rabbitmq/rabbitmq-server/assets/25414527/ebbd42cd-e1f0-4ae6-a0b5-9825cddc6c8d)
Any help is greatly appreciated!

**Reply by ioutkine** [reply]
I have sorted this out by restarting whole cluster. Could this be a bug? 
1. Set ```deprecated_features.permit.management_metrics_collection = false``` on one node
2. Restart the node. Metrics will be disabled now 
3. Set ```deprecated_features.permit.management_metrics_collection = true``` on the node
4. Restart the node. Metrics will not get enabled. To get it back you will need to stop other machines in the cluster

**Reply by christiansicari** [reply]
> I have sorted this out by restarting whole cluster. Could this be a bug?
> 
> 1. Set `deprecated_features.permit.management_metrics_collection = false` on one node
> 2. Restart the node. Metrics will be disabled now
> 3. Set `deprecated_features.permit.management_metrics_collection = true` on the node
> 4. Restart the node. Metrics will not get enabled. To get it back you will need to stop other machines in the cluster

How were you able to revert `deprecated_features.permit.management_metrics_collection = false`?
I've deployed Rabbit on EKS with the dedicated operator. I had the bad idea to set `deprecated_features.permit.management_metrics_collection = false`, but I can't revert it anymore.
I am stuck on this inconsistency
```console
rabbitmq@rabbitmq-server-0:/$ cat /etc/rabbitmq/conf.d/
10-defaults.conf                  10-operatorDefaults.conf          11-default_user.conf              90-userDefinedConfiguration.conf  
rabbitmq@rabbitmq-server-0:/$ cat /etc/rabbitmq/conf.d/* | grep management_metrics_collection
deprecated_features.permit.management_metrics_collection = true
rabbitmq@rabbitmq-server-0:/$ rabbitmqctl list_deprecated_features
Listing deprecated features ...
┌───────────────────────────────┬──────────────────────┬───────────┐
│ name                          │ deprecation_phase    │ state     │
├───────────────────────────────┼──────────────────────┼───────────┤
│ amqp_address_v1               │ permitted_by_default │ permitted │
├───────────────────────────────┼──────────────────────┼───────────┤
│ amqp_filter_set_bug           │ permitted_by_default │ permitted │
├───────────────────────────────┼──────────────────────┼───────────┤
│ classic_queue_mirroring       │ removed              │ denied    │
├───────────────────────────────┼──────────────────────┼───────────┤
│ global_qos                    │ permitted_by_default │ permitted │
├───────────────────────────────┼──────────────────────┼───────────┤
│ management_metrics_collection │ permitted_by_default │ denied    │
├───────────────────────────────┼──────────────────────┼───────────┤
│ queue_master_locator          │ permitted_by_default │ permitted │
├───────────────────────────────┼──────────────────────┼───────────┤
│ ram_node_type                 │ permitted_by_default │ denied    │
├───────────────────────────────┼──────────────────────┼───────────┤
│ transient_nonexcl_queues      │ permitted_by_default │ permitted │
└───────────────────────────────┴──────────────────────┴───────────┘
rabbitmq@rabbitmq-server-0:/$ 
```

**Reply by michaelklishin** [reply]
@christiansicari FYI rabbitmq/rabbitmq-server#12802, rabbitmq/rabbitmq-server#12840. The fix shipped in `4.1.0` and `4.0.5`.

I don't remember a false positive around  `deprecated_features.permit.management_metrics_collection` specifically. Now that some deprecated features are disabled by default in the upcoming `4.3.0` release, our team had to update dozens of test suites and in some, explicitly allow deprecated features. There were not fixes or surprises that I recall.

**Reply by michaelklishin** [reply]
There were no false positives for `management_metrics_collection`.
